### DISCUSSION TITLE
changing automated testing

### DISCUSSION THREAD

**Discussion by wingechr** [openego/oeplatform#1209] [Ideas]
A question for everyone:

The current testing setup is very slow and seemingly overly complex.

* In the [workflow](https://github.com/OpenEnergyPlatform/oeplatform/blob/develop/.github/workflows/automated-testing.yaml), we run everything twice (multiple versions of python), **including** the setup of the oedb
* we install all the python dependencies, but then use tox which installs all the python dependencies **again**.

## Proposal:

* running tests in different python versions is what tox is supposed to do
* so we remove the multiple workflow runs and instead do it in tox
* some of the steps (install requirement,txt, django migrate, ....) do not have to be run in the main workflow, because the test database is created when tox runs the actual django unit tests.

## Questions:

@Ludee @jh-RLI  @henhuy  @Darynarli 

do you see any problems with that?

**Comment by henhuy** [comment]
Seems reasonable
